### PR TITLE
Migrate logger from knative to controller-runtime

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

This updates from using the `knative.dev/pkg` logger to the controller-runtime-provided logger.

This is a buddy change to the upstream update of the logger: https://github.com/kubernetes-sigs/karpenter/pull/1050

**How was this change tested?**

`make presubmit`
`/karpenter snapshot`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: out.log
Comment (njtran) [inline out.log:1]: Did you mean for this to be in here?

File: pkg/batcher/createfleet.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func execCreateFleetBatch(ec2api ec2iface.EC2API) BatchExecutor[ec2.CreateFleetI
 			for _, instanceID := range reservation.InstanceIds {
 				requestIdx++
 				if requestIdx >= len(inputs) {
-					logging.FromContext(ctx).Errorf("received more instances than requested, ignoring instance %s", aws.StringValue(instanceID))
+					log.FromContext(ctx).Error(fmt.Errorf("received more instances than requested, ignoring instance %s", aws.StringValue(instanceID)), "Batcher error")
```
Comment (njtran) [inline pkg/batcher/createfleet.go:73]: I had similar comments on the upstream PR, but I think we should aim to align these with the rest of our logging semantics of noncapitalized
```suggestion
					log.FromContext(ctx).Error(fmt.Errorf("received more instances than requested, ignoring instance %s", aws.StringValue(instanceID)), "batcher error")
```

File: pkg/batcher/describeinstances.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func (b *DescribeInstancesBatcher) DescribeInstances(ctx context.Context, descri
 func FilterHasher(ctx context.Context, input *ec2.DescribeInstancesInput) uint64 {
 	hash, err := hashstructure.Hash(input.Filters, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
 	if err != nil {
-		logging.FromContext(ctx).Errorf("error hashing")
+		log.FromContext(ctx).Error(fmt.Errorf("error hashing"), "Batcher error")
```
Comment (njtran) [inline pkg/batcher/describeinstances.go:59]: same here

File: pkg/controllers/interruption/controller.go [hunk 3]
```diff
@@ -105,7 +105,7 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 		msg, e := c.parseMessage(sqsMessages[i])
 		if e != nil {
 			// If we fail to parse, then we should delete the message but still log the error
-			logging.FromContext(ctx).Errorf("parsing message, %v", e)
+			log.FromContext(ctx).Error(fmt.Errorf("parsing message, %w", err), "Interruption error")
```
Comment (njtran) [inline pkg/controllers/interruption/controller.go:108]:
```suggestion
			log.FromContext(ctx).Error(fmt.Errorf("parsing message, %w", err), "interruption error")
```

File: pkg/batcher/terminateinstances.go [hunk 4]
```diff
@@ -68,7 +68,7 @@ func execTerminateInstancesBatch(ec2api ec2iface.EC2API) BatchExecutor[ec2.Termi
 		// We don't care about the error here since we'll break up the batch upon any sort of failure
 		output, err := ec2api.TerminateInstancesWithContext(ctx, firstInput)
 		if err != nil {
-			logging.FromContext(ctx).Errorf("terminating instances, %s", err)
+			log.FromContext(ctx).Error(fmt.Errorf("terminating instances, %w", err), "Batcher error")
```
Comment (njtran) [inline pkg/batcher/terminateinstances.go:71]:
```suggestion
			log.FromContext(ctx).Error(fmt.Errorf("terminating instances, %w", err), "batcher error")
```

File: pkg/controllers/nodeclaim/tagging/controller.go [hunk 5]
```diff
@@ -54,18 +54,18 @@ func NewController(kubeClient client.Client, instanceProvider instance.Provider)
...
 	if !isTaggable(nodeClaim) {
 		return reconcile.Result{}, nil
 	}
-	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("provider-id", nodeClaim.Status.ProviderID))
+	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithValues("provider-id", nodeClaim.Status.ProviderID))
 	id, err := utils.ParseInstanceID(nodeClaim.Status.ProviderID)
 	if err != nil {
 		// We don't throw an error here since we don't want to retry until the ProviderID has been updated.
-		logging.FromContext(ctx).Errorf("failed to parse instance ID, %w", err)
+		log.FromContext(ctx).Error(fmt.Errorf("parsing instance ID, %w", err), "Tagging error")
```
Comment (njtran) [inline pkg/controllers/nodeclaim/tagging/controller.go:68]:
```suggestion
		log.FromContext(ctx).Error(fmt.Errorf("parsing instance ID, %w", err), "tagging error")
```

File: pkg/providers/instance/instance.go [hunk 6]
```diff
@@ -212,7 +213,7 @@ func (p *DefaultProvider) launchInstance(ctx context.Context, nodeClass *v1beta1
 		return nil, fmt.Errorf("getting launch template configs, %w", err)
 	}
 	if err := p.checkODFallback(nodeClaim, instanceTypes, launchTemplateConfigs); err != nil {
-		logging.FromContext(ctx).Warn(err.Error())
+		log.FromContext(ctx).Error(err, "CloudProvider error")
```
Comment (njtran) [inline pkg/providers/instance/instance.go:216]:
```suggestion
		log.FromContext(ctx).Error(err, "cloudprovider error")
```

File: pkg/providers/instancetype/instancetype.go [hunk 7]
```diff
@@ -267,7 +267,7 @@ func (p *DefaultProvider) createOfferings(ctx context.Context, instanceType *ec2
 				// ignore since karpenter doesn't support it yet, but do not log an unknown capacity type error
 				continue
 			default:
-				logging.FromContext(ctx).Errorf("Received unknown capacity type %s for instance type %s", capacityType, *instanceType.InstanceType)
+				log.FromContext(ctx).WithValues("capacity-type", capacityType, "instance-type", *instanceType.InstanceType).Error(fmt.Errorf("received unknown capacity type"), "CloudProvider error")
```
Comment (njtran) [inline pkg/providers/instancetype/instancetype.go:270]:
```suggestion
				log.FromContext(ctx).WithValues("capacity-type", capacityType, "instance-type", *instanceType.InstanceType).Error(fmt.Errorf("received unknown capacity type"), "cloudprovider error")
```

**Review by njtran — COMMENTED** [review]
Looks like you missed some 

```
./pkg/controllers/nodeclass/termination/controller.go:	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("nodeclass.termination").With("ec2nodeclass", nodeClass.Name))
./pkg/controllers/nodeclass/hash/controller.go:	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("nodeclass.hash").With("ec2nodeclass", nodeClass.Name))
./pkg/controllers/nodeclass/status/controller.go:	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("nodeclass.status").With("ec2nodeclass", nodeClass.Name))
```

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by njtran — APPROVED** [review]
LGTM 🚢

**Pull request merged**